pkg/frontend/v2/frontend.go

```diff
@@ -205,6 +218,18 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 			retryOnTooManyOutstandingRequests: f.cfg.RetryOnTooManyOutstandingRequests && f.schedulerWorkers.getWorkersCount() > 1,
 		}
 
+		queryPriority := f.limits.QueryPriority(userID)
+
+		if reqParams != nil && queryPriority.Enabled {
+			queryPriorityChanged := !reflect.DeepEqual(f.queryPriority, queryPriority)
+			if queryPriorityChanged {
+				f.queryPriority = queryPriority
+				f.compiledQueryPriority = queryPriority
```

Comment:
Reading and writing `f.queryPriority` and `f.compiledQueryPriority` seems requires lock protection.

Reply:
Great call! Addressed in https://github.com/cortexproject/cortex/pull/5605/commits/96ff019ddba591815be280b4f9c9aacf63733a42